Create 2025-08-08-july-2025-roundup.md (#595)
memfault/interrupt: _drafts/2025-08-08-july-2025-roundup.md

A super fun deep dive into undocumented features of the Microchip VSC8512 PHY. – Noah

- [**STM32L431 Teardown – Die Shots and Analysis**](https://serd.es/2025/01/02/STM32L431-teardown.html)  
Detailed teardown and silicon analysis of the STM32L431. – Noah

- [**New Chapter for IAR Docker Support: Introducing the IAR Cloud-Ready Containers Repository**](https://www.iar.com/blog/a-new-chapter-for-our-docker-support-introducing-the-iar-cloud-ready-containers-repository)  
I like to see IAR continuing to invest in making it easier to test their stuff in CI. Has always been a bit of a pain point. – Chris

- [**Efficient Delta Updates for Embedded Linux: Rugix**](https://oss.silitics.com/rugix/blog/efficient-delta-updates/)  
Compares OTA delta update strategies with real-world benchmarks. Super useful for teams managing update pipelines. – Pat

- [**How to Write Rust in the Kernel: Part 1**](https://lwn.net/Articles/1024202/)  
A practical, well-explained intro to Rust in the Linux kernel. – Noah

- [**Efficient On‑Device Core Dump Processing for IoT**](https://www.youtube.com/watch?v=fDwDXg7T4K8)  

> Blake's EOSS talk is now on YouTube! He walks through how Rust can enable on-device core dump processing without offloading. – Pat

Reviewer: ```suggestion
Blake Hildebrand (from Memfault) gave an EOSS talk that is now on YouTube! He walks through how Rust can enable on-device core dump processing without offloading. – Pat
```

- [**Pre-Main Zephyr: Part 1**](https://n-eq.github.io/blog/2025/07/03/pre-main-zephyr-1)  

> Interrupt community member Nabil Elqatib dives into Zephyr’s startup process on ARM Cortex-M CPUs in this great three-part series ([Part 2](https://n-eq.github.io/blog/2025/07/12/pre-main-zephyr-2), [Part 3](https://n-eq.github.io/blog/2025/07/21/pre-main-zephyr-3)). – Gilly

Reviewer: Let's take out attribution for me since I'm technically the author 😄  + a few other edits
```suggestion
Interrupt community member Nabil Elqatib dives into Zephyr’s startup process on ARM Cortex-M CPUs. See [Part 2](https://n-eq.github.io/blog/2025/07/12/pre-main-zephyr-2) and [Part 3](https://n-eq.github.io/blog/2025/07/21/pre-main-zephyr-3)) as well!
```

Author: Oh, right, I don't know why I was doing that! Sorry, brain fart - thank you for catching!

## Projects & Tools

- [**sequential-storage (Rust crate)**](https://crates.io/crates/sequential-storage)  
Interesting library providing wear-leveled object storage in flash. – Noah

- [**RP2350 A4, RP2354, and a New Hacking Challenge – Raspberry Pi**](https://www.raspberrypi.com/news/rp2350-a4-rp2354-and-a-new-hacking-challenge/)  
Erratum A9 fixed. – Noah

- [**Hardware-Aware Agent: Generate Production-Level Firmware in Seconds – Embedder.dev**](https://embedder.dev/)  
Another attempt at datasheet → peripheral driver coding assistant. – Chris

## News & Announcements

- [**Qt to Acquire IAR Systems**](https://www.qt.io/stock/inside-information-qts-recommended-public-cash-offer-for-iar-systems-group-1751607060000-3623512)  

> Noteworthy industry consolidation in the embedded tooling space. – Gilly 

Reviewer: ```suggestion
Noteworthy acquisition in the embedded industry!
```

- [**Checking Out sys.remote_exec in Python 3.14**](https://rtpg.co/2025/06/28/checking-out-sys-remote-exec/)  
Awesome feature coming in Πython 3.14. – Noah

- [**JLCPCB Launches Transparent FPC**](https://jlcpcb.com/blog/jlcpcb-launches-transparent-fpc-see-innovation-in-a-new-light)  
JLC PCB has [transparent PCBs](https://hackaday.com/2025/07/25/transparent-pcbs-trigger-90s-nostalgia/) now! – Noah  

- [**Zephyr 4.2.0 is Now Available**](https://www.zephyrproject.org/zephyr-rtos-4-2-now-available-introduces-renesas-rx-support-usb-video-class-and-more/)  

> Zephyr 4.2.0 is out, with support for Renesas RX, USB Video Class, and 96 new boards. – Gilly

Reviewer: Same as above
```suggestion
Zephyr 4.2.0 is out, with support for Renesas RX, USB Video Class, and 96 new boards.
```

- [**Linux 6.16 Release Notes**](https://lwn.net/Articles/970157/)  

> Linux 6.16 adds Intel APX, USB audio offload, zero-copy TCP via DMABUF, and early support for NVIDIA Blackwell. – Gilly

Reviewer: ```suggestion
Linux 6.16 adds Intel APX, USB audio offload, zero-copy TCP via DMABUF, and early support for NVIDIA Blackwell.
```

> - [**Linux 6.16 Release Notes**](https://lwn.net/Articles/970157/)  
> Linux 6.16 adds Intel APX, USB audio offload, zero-copy TCP via DMABUF, and early support for NVIDIA Blackwell. – Gilly
> 
> 

Reviewer: I don't think this is a major release, but let's keep an eye out for more!
```suggestion
```

## Upcoming Events

- [**Tuesday, August 19 – Coredump EP #015: Developing Kid-Safe Tech at Gabb: What It Takes and Why It’s So Important**](https://memfault.com/resources/developing-kid-safe-tech-at-gabb-what-it-takes-and-why-its-so-important/)<br>

> When you're building tech for kids, reliability isn’t negotiable. Devices need to be always-on, always-connected, and dead-simple to use. Gabb’s engineering team joins Memfault to share how they tackle power constraints, OTA updates, and safety-first design. Join us live at 8AM PT | 11AM ET | 5PM CET.

Reviewer: What do you think of simplifying this announcement for the interrupt audience?
```suggestion
Gabb’s engineering team joins Memfault to share how they tackle power constraints, OTA updates, and safety-first design. Join us live at 8AM PT | 11AM ET | 5PM CET.
```

- [**Tuesday, September 16 – Coredump EP #016: From Startup to Global Brand: Scaling Engineering at reMarkable**](https://memfault.com/resources/from-startup-to-global-brand-scaling-engineering-at-remarkable/)<br>

> How do you grow from a startup with homegrown systems to a global brand with millions of devices without compromising quality? reMarkable CTO Nico Cormier joins Memfault’s François Baldassari and Chris Coleman to share hard-won lessons on release velocity, operational stability, and scaling engineering teams. Join us live at 8AM PT | 11AM ET | 5PM CET.

Reviewer: Same idea here:
```suggestion
reMarkable CTO Nico Cormier joins Memfault’s François Baldassari and Chris Coleman to share lessons on release velocity, operational stability, and scaling engineering teams. Join us live at 8AM PT | 11AM ET | 5PM CET.
```

- [**September 16–November 4: Nordic Tech Tour, Americas 2025**](https://www.nordicsemi.com/Events/2025/Nordic-Tech-Tour-Americas-2025)<br>

> Nordic is hitting the road this fall with 14 in-person events across the Americas. Join wireless IoT experts—and the Memfault team—for hands-on workshops and sessions covering Bluetooth LE, satellite and cellular IoT, power management, and the latest nRF54 Series SoCs. Free to attend, with limited workshop seats available.

Reviewer: From what I understand, we won't be able to make all of them. Should we remove the Memfault team callout here?
```suggestion
Nordic is hitting the road this fall with 14 in-person events across the Americas. Join wireless IoT experts for hands-on workshops and sessions covering Bluetooth LE, satellite and cellular IoT, power management, and the latest nRF54 Series SoCs. Free to attend, with limited workshop seats available.
```
